File: split.js
const fs = require('fs');

let content = fs.readFileSync('3ddesignstudio (2).html', 'utf-8');

// Extract all scripts
const scriptRegex = /<script[^>]*>(.*?)<\/script>/gis;
let js_content = "";
let match;
while ((match = scriptRegex.exec(content)) !== null) {
    const fullTag = match[0].toLowerCase();
    if (!fullTag.includes('ld+json') && !fullTag.includes('cloudflare')) {
        js_content += match[1] + "\n";
    }
}

// Remove all <style> blocks
let clean_html = content.replace(/<style.*?>.*?<\/style>/gis, '');

// Remove all <script> blocks (except cloudflare and ld+json)
clean_html = clean_html.replace(/<script[^>]*>.*?<\/script>/gis, (match) => {
    const lowerMatch = match.toLowerCase();
    if (lowerMatch.includes('ld+json') || lowerMatch.includes('cloudflare')) {
        return match;
    }
    return '';
});

// Inject our links into HEAD and BODY
clean_html = clean_html.replace(/<\/head>/i, '<link rel="stylesheet" href="css/style.css">\n</head>');
clean_html = clean_html.replace(/<\/body>/i, '<script src="components/header.js"></script>\n<script src="components/footer.js"></script>\n<script src="js/main.js" defer></script>\n</body>');

fs.writeFileSync('js/main.js', js_content, 'utf-8');
fs.writeFileSync('index.html', clean_html, 'utf-8');

console.log('Split successful.');

File: split.py
import re

with open('3ddesignstudio (2).html', 'r', encoding='utf-8') as f:
    content = f.read()

# Extract all scripts
scripts = re.findall(r'<script[^>]*>(.*?)</script>', content, flags=re.DOTALL | re.IGNORECASE)

js_content = ""
for s in scripts:
    if "ld+json" not in s and "email-decode" not in s:
        js_content += s + "\n"

# Remove all <style> blocks
clean_html = re.sub(r'<style.*?>.*?</style>', '', content, flags=re.DOTALL | re.IGNORECASE)

# Remove all <script> blocks (except cloudflare and ld+json)
def script_repl(match):
    tag = match.group(0)
    if 'ld+json' in tag or 'cloudflare' in tag:
        return tag
    return ''

clean_html = re.sub(r'<script[^>]*>.*?</script>', script_repl, clean_html, flags=re.DOTALL | re.IGNORECASE)

# Inject our links into HEAD and BODY
clean_html = clean_html.replace('</head>', '<link rel="stylesheet" href="css/style.css">\n</head>')
clean_html = clean_html.replace('</body>', '<script src="components/header.js"></script>\n<script src="components/footer.js"></script>\n<script src="js/main.js" defer></script>\n</body>')

with open('js/main.js', 'w', encoding='utf-8') as f:
    f.write(js_content)

with open('index.html', 'w', encoding='utf-8') as f:
    f.write(clean_html)

print("Split successful.")
